crates/router/src/core/payment_methods/cards.rs

```diff
@@ -108,7 +114,24 @@ pub async fn add_payment_method(
             Ok((payment_method_response, false))
         }
     };
-    Ok(response?.0).map(services::ApplicationResponse::Json)
+
+    let (resp, is_duplicate) = response?;
+    if !is_duplicate {
+        let pm_metadata = resp.metadata.as_ref().map(|data| data.clone().expose());
```

Comment:
```suggestion
        let pm_metadata = resp.metadata.as_ref().map(|data| data.peek());
```

and use 
`pm_metadata.cloned()` in line 127